File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.cc
#include <iostream>
#define fio ios_base::sync_with_stdio(0), cin.tie(0), cout.tie(0);
#define pii pair<int, int>

using namespace std;

const int MAX = 101;
int N, K;
int sw[MAX];
pii student[MAX];

void input()
{
    cin >> N;
    for (int i = 1; i <= N; i++) {
        cin >> sw[i];
    }

    cin >> K;
    for (int i = 0; i < K; i++) {
        cin >> student[i].first >> student[i].second;
    }
}

void toggle(int gen, int num)
{
    if (gen == 1) {
        for (int i = num; i <= N; i += num) {
            sw[i] ^= 1;
        }
    }
    else {
        int idx = num;
        sw[idx] ^= 1;
        int cnt = 1;
        while (1) {
            if (idx - cnt <= 0 || idx + cnt > N)
                break;
            if (sw[idx + cnt] == sw[idx - cnt]) {
                sw[idx + cnt] ^= 1;
                sw[idx - cnt] ^= 1;
                cnt++;
            }
            else
                break;
        }
    }
}

void sol() {
    int gen, num;
    for (int i = 0; i < K; i++) {
        gen = student[i].first;
        num = student[i].second;

        toggle(gen, num);
    }

    for (int i = 1; i <= N; i++) {
        cout << sw[i] << " ";
        if (i % 20 == 0)
            cout << "\n";
    }
}

void run() {
    input();
    sol();
}

int main() {
    fio;
    run();
    return 0;
}

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static final int MAX = 101;
    static int N;
    static int[] arr = new int[MAX];

    public static void main(String[] args) throws Exception {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        N = Integer.parseInt(in.readLine());
        st = new StringTokenizer(in.readLine());
        for (int i = 1; i <= N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int M = Integer.parseInt(in.readLine());
        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(in.readLine());
            int gen = Integer.parseInt(st.nextToken());
            int num = Integer.parseInt(st.nextToken());

            toggle(gen, num);
        }
        print();
    }

    static void toggle(int gen, int num) {
        if (gen == 1) {
            for (int i = num; i <= N; i += num) {
                arr[i] ^= 1;
            }
        } else {
            int idx = num;
            arr[idx] ^= 1;
            int cnt = 1;
            while (true) {
                if (idx - cnt <= 0 || idx + cnt > N)
                    break;
                if (arr[idx + cnt] == arr[idx - cnt]) {
                    arr[idx + cnt] ^= 1;
                    arr[idx - cnt] ^= 1;
                    cnt++;
                } else
                    break;
            }
        }
    }

    static void print() {
        for (int i = 1; i <= N; i++) {
            System.out.printf("%d ", arr[i]);
            if (i % 20 == 0)
                System.out.println();
        }
    }
}